internal/kafka/internal/api/dbapi/kafka_request_types.go

```diff
@@ -261,3 +263,46 @@ func (k *KafkaRequest) IsUsingSharedTLSCertificate(kafkaConfig *config.KafkaConf
 func (k *KafkaRequest) IsADeveloperInstance() bool {
 	return k.InstanceType == types.DEVELOPER.String()
 }
+
+//////////////////////////////////
+/// functional builder
+
+type kafkaRequestFunctionalParam func(request *KafkaRequest)
+
+// makeFunc creates a functional parameter for the specified struct field and with the specified value type
+func makeFunc[T any](dest string) func(value T) kafkaRequestFunctionalParam {
+	return func(value T) kafkaRequestFunctionalParam {
+		return func(request *KafkaRequest) {
+			r := reflect.ValueOf(request)
+			reflect.Indirect(r).FieldByName(dest).Set(reflect.ValueOf(value))
+		}
+	}
+}
+
+var WithID = makeFunc[string]("ID")
```

Comment:
We should consider avoiding global variables where possible.
Could we move this to specific unexported test related methods?

Reply:
No. It is a function and it is used in a different package.

Reply:
The returned value is a function but it is stored in a global variable

Reply:
It is stored in a global variable of type `function` and is meant to be used as `function` in tests outside of this package. I could move it to a 'test` package, but it needs to be visible outside the package.

Reply:
moving it to a test package makes sense. Or renaming it in a way that it is clear that it is used for testing purposes.

> It is stored in a global variable of type function and is meant to be used as function in tests outside of this package.

That still doesn't require us to put it in a global variable. It could be in a function itself

Reply:
> As a side note, have we considered to create it on the package where the tests are located?

Yes. There are many tests where we create KafkaRequests, I believe the functional builder can be useful in more than just one test.